pkg/operator/controller/ingress/deployment.go

```diff
@@ -336,7 +341,49 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 	secretName := controller.RouterEffectiveDefaultCertificateSecretName(ci, deployment.Namespace)
 	deployment.Spec.Template.Spec.Volumes[0].Secret.SecretName = secretName.Name
 
+	var (
+		accessLogging  *operatorv1.AccessLogging
+		accessLogLevel = "debug"
+	)
 	if enabled, level := ExtraLoggingEnabled(ci, ingressConfig); enabled {
+		accessLogging = &operatorv1.AccessLogging{
+			Destination: operatorv1.LoggingDestination{
+				Type:      operatorv1.ContainerLoggingDestinationType,
+				Container: &operatorv1.ContainerLoggingDestinationParameters{},
+			},
+		}
+		accessLogLevel = level
+	} else if ci.Spec.Logging != nil && ci.Spec.Logging.Access != nil {
+		switch ci.Spec.Logging.Access.Destination.Type {
+		case operatorv1.ContainerLoggingDestinationType:
+			accessLogging = &operatorv1.AccessLogging{
+				Destination: operatorv1.LoggingDestination{
+					Type:      operatorv1.ContainerLoggingDestinationType,
+					Container: &operatorv1.ContainerLoggingDestinationParameters{},
+				},
+				HttpLogFormat: ci.Spec.Logging.Access.HttpLogFormat,
+			}
+		case operatorv1.SyslogLoggingDestinationType:
+			if ci.Spec.Logging.Access.Destination.Syslog != nil {
```

Comment:
Would a `nil` field be a valid state in this case? It looks like if type is Syslog and `Spec.Logging.Access.Destination.Syslog` is nil, we do nothing and also don't reject the IngressController as invalid? Or does openapi handle this case somehow via the union type?

Reply:
I'll check whether specifying a nil value passes validation.  If it does, then yeah, the user can specify `spec.logging.access.destination.type: Syslog` and a nil value for `spec.logging.access.destination.syslog`, and the result will be that logging will not be enabled.  Without `spec.logging.access.destination.syslog`, we don't have an endpoint.

Reply:
Yup, I can create an ingresscontroller with `spec.logging.access.destination.type: Syslog` and without `spec.logging.access.destination.syslog`.